### PR TITLE
Delete torchtune's inplace copy definition for NF4

### PR DESCRIPTION
**Pull request by ebsmothers**
Thanks to @msaroufim for root causing this tricky bug. This addresses https://github.com/pytorch/ao/issues/642 on our end as well as #1246.

**Update**: Instead of reenabling our inplace copy, @gau-nernst has found the upstream fix in torchao (described [here](https://github.com/pytorch/ao/issues/642#issuecomment-2306089086)). With [this fix](https://github.com/pytorch/ao/pull/737), we can fully delete the inplace copy from torchtune. Until we're on the ao version with the fix our QLoRA will continue to be slow, but after that things should be back to normal.

###Stuff below this line is outdated but kept as background###

We were defining our own override of aten copy op in torchtune prior to torchao version 0.2, when they added their own. After that we version-gated this override to ao < 0.2. But it turns out that the version we have is faster. Tbh idk why that is yet, but it should be safe to re-enable this to get us back to our previous perf on QLoRA state dict load. 

Tested by logging time from recipe startup to the beginning of the train loop.

On main: `init time: 101.48360734200105`
On this PR: `init time: 15.445241551031359`

### CODE REVIEW

File: torchtune/modules/low_precision/_register_nf4_dispatch_ops.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 
 import torch
 from torchao.dtypes.nf4tensor import implements as nf4_tensor_impl, to_nf4
-from torchtune.modules.low_precision._utils import _get_torchao_version
```
Comment (ebsmothers) [inline torchtune/modules/low_precision/_register_nf4_dispatch_ops.py:9]: Strictly speaking this util is now no longer used anywhere in our library, but I am inclined to keep it in for possible (likely?) future usage

**Comment by ebsmothers** [comment]
@weifengpy seems we may need to add an override for `aten.to.dtype_layout` to handle [this line](https://github.com/pytorch/pytorch/blob/da65cfbdea4f1f2176f6242004bda940a24f9ddb/torch/distributed/_composable/fsdp/_fsdp_param.py#L344) when creating FSDP params. Please lmk if I have the right understanding here

**Comment by weifengpy** [comment]
> @weifengpy seems we may need to add an override for `aten.to.dtype_layout` to handle [this line](https://github.com/pytorch/pytorch/blob/da65cfbdea4f1f2176f6242004bda940a24f9ddb/torch/distributed/_composable/fsdp/_fsdp_param.py#L344) when creating FSDP params. Please lmk if I have the right understanding here

yes, the error msg looks relevant. but I do not fully understand why torchtune implementation dispatch aten.to.dtype_layout but torchao’s implementation does not need it

FSDP2 needs to support around 9 tensor ops but aten.to.dtype_layout is totally new

File: torchtune/modules/low_precision/_register_nf4_dispatch_ops.py (hunk 1 above)
Comment (joecummings) [reply torchtune/modules/low_precision/_register_nf4_dispatch_ops.py:9]: Nope - get rid of it until we need it again. It's easy enough to find in git history.

**Review by joecummings — APPROVED** [review]

**Comment by weifengpy** [comment]
I see the PR is accepted. curious how did we resolve the local_tensor=nan issue?

**Review by msaroufim — APPROVED** [review]
very nice! glad the version check code is also gone 💀

**Pull request merged**